apps/native/app/src/screens/app-lock/app-lock.tsx

```diff
@@ -141,7 +142,8 @@ export const AppLockScreen: NavigationFunctionComponent<{
   useEffect(() => {
     setInvalidCode(false)
     if (code.length === MAX_PIN_CHARS) {
-      if (attempts === MAX_ATTEMPTS) {
+      // pinTries (attempts) starts at 0, so we need to subtract 1 to get the correct number of attempts
+      if (attempts === MAX_ATTEMPTS - 1) {
```

Comment:
1. Let's use `>=` comparisons to be safe (eg if someone had tried 3 times before upgrading the app).
2. Reviewing this logic, it looks like the final pin input isn't even tested. Should move this attempt + logout logic to when the codes don't match (line 165)